pyccel/ast/type_annotations.py

```diff
@@ -348,3 +334,18 @@ def __eq__(self, o):
                     self.order == o.order
         else:
             return False
+
+#==============================================================================
+
+typenames_to_dtypes = { 'float'   : PythonNativeFloat(),
+                        'double'  : PythonNativeFloat(),
+                        'real'    : PythonNativeFloat(),
+                        'complex' : PythonNativeComplex(),
+                        'int'     : PythonNativeInt(),
+                        'integer' : PythonNativeInt(),
```

Comment:
And do we still want to support `'integer'`?

Reply:
Similar answer to above. Definitely not in this PR. In general, I'm not sure. It is less strange than "real"

Reply:
We could add it here: https://github.com/pyccel/pyccel/issues/1786

Reply:
I think we should remove it, and yes this could go in the same issue as `'real'`.

Reply:
I have updated the issue description